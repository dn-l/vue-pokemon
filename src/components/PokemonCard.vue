<template>
  <v-card
      raised
      class="ma-4"
      width="15em"
      :loading="pokemon.isLoading"
    >
      <v-img
        height="15em"
        position="top"
        :src="pokemon.image"
      >
      </v-img>
      <v-card-title>{{pokemon.name || '&nbsp;'}}</v-card-title>
      <v-card-actions>
        <v-btn @click="toggleFavourite" icon>
          <v-icon v-if="isFavourite">mdi-heart</v-icon>
          <v-icon v-else>mdi-heart-outline</v-icon>
        </v-btn>
        <div class="flex-grow-1"></div>
        <v-btn outlined :to="learnMoreRoute">
          Learn more
        </v-btn>
      </v-card-actions>
    </v-card>
</template>

<script lang="ts">
import Vue from 'vue';
import Pokemon from './PokemonList/Pokemon';

export default Vue.extend({
  props: ['pokemon'],
  methods: {
    toggleFavourite() {
      this.$store.dispatch('toggleFavourite', this.$props.pokemon.id);
    },
  },
  computed: {
    isFavourite() {
      return this.$store.getters.isFavourite(this.pokemon.id);
    },
    learnMoreRoute() {
      const pokemon = this.pokemon as Pokemon;
      if (pokemon.isLoading) {
        return {};
      }
      return {
        name: 'pokemon',
        params: { id: pokemon.id },
      };
    },
  },
});
</script>
