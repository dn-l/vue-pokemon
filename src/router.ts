import Vue from 'vue';
import Router from 'vue-router';
import Pokemons from './views/Pokemons.vue';

export const routes = [
  {
    path: '/',
    name: 'pokemons',
    component: Pokemons,
  },
  {
    path: '/pokemon/:id',
    name: 'pokemon',
    component: () => import(/* webpackChunkName: "pokemon" */ './views/Pokemon.vue'),
  },
  {
    path: '*',
    redirect: '/',
  },
];

Vue.use(Router);

export default new Router({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
});
