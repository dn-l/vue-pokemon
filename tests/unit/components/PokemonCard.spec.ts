import Vue from 'vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { routes } from '@/router';
import PokemonCard from '@/components/PokemonCard.vue';

const pokemon = {
  id: '1',
  name: 'Bulbasaur',
  image: 'https://img.pokemondb.net/artwork/bulbasaur.jpg',
};

const localVue = createLocalVue();
Vue.use(Vuetify);
localVue.use(Vuex);
localVue.use(VueRouter);

const router = new VueRouter({ routes });

const store = new Vuex.Store({
  getters: {
    isFavourite() {
      return () => false;
    },
  },
});

describe('PokemonCard.vue', () => {
  const wrapper = shallowMount(PokemonCard, {
    propsData: { pokemon },
    store,
    localVue,
    router,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders name', () => {
    expect(wrapper.text()).toMatch(pokemon.name);
  });
});
