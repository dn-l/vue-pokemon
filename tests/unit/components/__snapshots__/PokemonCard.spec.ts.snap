// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard.vue matches snapshot 1`] = `
<v-card-stub loaderheight="4" tag="div" width="15em" raised="true" class="ma-4">
  <v-img-stub height="15em" position="top" src="https://img.pokemondb.net/artwork/bulbasaur.jpg" transition="fade-transition"></v-img-stub>
  <v-card-title-stub>Bulbasaur</v-card-title-stub>
  <v-card-actions-stub>
    <v-btn-stub tag="button" activeclass="" icon="true" type="button">
      <v-icon-stub>mdi-heart-outline</v-icon-stub>
    </v-btn-stub>
    <div class="flex-grow-1"></div>
    <v-btn-stub tag="button" activeclass="" to="[object Object]" outlined="true" type="button">
      Learn more
    </v-btn-stub>
  </v-card-actions-stub>
</v-card-stub>
`;
